File: backend/src/main/java/com/g4ts/agendaapp/model/Proyecto.java
package com.g4ts.agendaapp.model;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "Proyecto")
@Getter @Setter @ToString @Builder
@AllArgsConstructor
@NoArgsConstructor
public class Proyecto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String nombre;
    private String descripcion;
    private LocalDate fechaInicio;
    private LocalDate fechaPrevistaFin;
    private String ubicacion;

    @OneToOne
    @JoinColumn(name = "idUsuario")
    private Usuario usuario;
}

File: frontend/src/app/model/proyecto.ts
import { Usuario } from "./usuario";

export class Proyecto {

    id: String;
    nombre: String;
    descripcion: String;
    fechaInicio: String;
    fechaPrevistaFin: String;
    ubicacion: String;
    visibilidad: String;
    usuario: Usuario;

    constructor() {
        this.usuario = new Usuario();
    }

}
